module/zfs/zfs_dir.c

```diff
@@ -508,8 +513,51 @@ zfs_unlinked_drain(zfsvfs_t *zfsvfs)
 
 		zp->z_unlinked = B_TRUE;
 		iput(ZTOI(zp));
```

Comment:
cool, I didn't realize that.

Reply:
Thanks.  Canceling would be nice, but even without canceling this is an improvement over the current method, and it would be good enough nearly all the time (i.e. where you aren't trying to unmount it before the queue is drained).  AFAIK the typical use case is that you delete a large file, and then the system crashes and when you come back up, the mount "hangs" for hours while the large file reclamation completes.